Dockerfile

```
@@ -1,12 +1,31 @@
+# Copy entrypoint script
+COPY entrypoint.sh /app/entrypoint.sh
+RUN chmod +x /app/entrypoint.sh
+
+# Cloud Run sets the PORT environment variable
 ENV PORT 8080
 
-CMD ["uvicorn", "main:app", "--host", "0.0.0.0", "--port", "8080"]
+ENTRYPOINT ["/app/entrypoint.sh"]
+
+# Start uvicorn
+CMD uvicorn app.main:app --host 0.0.0.0 --port ${PORT:-8080}
```

Hadolint (2.14.0)

[warning] Use arguments JSON notation for CMD and ENTRYPOINT arguments

(DL3025)